processing_scripts/Workflow7_ImageJ_Jython_Analyst.py

```diff
@@ -0,0 +1,236 @@
+    pixType = FormatTools.pixelTypeFromString(pixtype)
+    bpp = FormatTools.getBytesPerPixel(pixType)
+    isSigned = FormatTools.isSigned(pixType)
+    isFloat = FormatTools.isFloatingPoint(pixType)
+    isLittle = False
+    interleave = False
+    
+    store = gateway.getPixelsStore(ctx)
+    pixelsId = pixels.getId()
+    store.setPixelsId(pixelsId, False)
+    stack = ImageStack(sizeX, sizeY)
+    for t in range(0,sizeT):
+        for z in range(0,sizeZ):
+            for c in range(0, sizeC):
+                plane = store.getPlane(z, c, t)
+
+                channel = ImageTools.splitChannels(plane, 0, 1, bpp, False, interleave)
+                pixels = DataTools.makeDataArray(plane, bpp, isFloat, isLittle)
+
+                q = pixels
+                if (len(plane) != sizeX*sizeY):
+                    tmp = q
+                    q = zeros(sizeX*sizeY, 'h')
+                    System.arraycopy(tmp, 0, q, 0, Math.min(len(q), len(tmp)))
+                    if isSigned:
+                        q = DataTools.makeSigned(q)
+                    
+                if q.typecode == 'b':
+                    ip = ByteProcessor(sizeX, sizeY, q, None)
+                elif q.typecode == 'h':
+                    ip = ShortProcessor(sizeX, sizeY, q, None)
+                stack.addSlice('', ip)
+    # Do something
+    imp = ImagePlus(image.getName(), stack)
+    imp.setDimensions(sizeC, sizeZ, sizeT)
+    imp.setOpenAsHyperStack(True)
+    imp.show()
+    return imp
+
+# upload image to OMERO server
+def upload_image(path, gateway, id):
+    "Upload an image to omero"
+
+    user = gateway.getLoggedInUser()
+    sessionKey = gateway.getSessionId(user)
+
+    config = ImportConfig()
+    config.debug.set('false')
+    config.hostname.set(HOST)
+    config.sessionKey.set(sessionKey)
+    value = "omero.model.Dataset:"
+    value += str(id)
+    config.target.set(value)
+
+    loci.common.DebugTools.enableLogging("DEBUG")
+
+    store = config.createStore()
+    reader = OMEROWrapper(config)
+
+    library = ImportLibrary(store, reader)
+    error_handler = ErrorHandler(config)
+
+    library.addObserver(LoggingImportMonitor())
+    candidates = ImportCandidates(reader, path, error_handler)
+    reader.setMetadataOptions(DefaultMetadataOptions(MetadataLevel.ALL))
+    return library.importCandidates(config, candidates)
+
+
+# Prototype analysis example
+gateway = connect_to_omero()
+ctx = SecurityContext(group_id)
+exp = gateway.getLoggedInUser()
+exp_id = exp.getId()
+
+# get all images_ids in an omero dataset
+dataset_id = 1107
+ids = get_image_ids(gateway, ctx, dataset_id)
+
+#if target_user ~= None:
+# Switch context to target user and open omeroImage as ImagePlus object
+ctx = switchSecurityContext(ctx, target_user)
+imp = openOmeroImage(ctx, image_id)
+
+#Some analysis which creates ROI's and Results Table
+IJ.setAutoThreshold(imp, "Default dark")
+IJ.run(imp, "Analyze Particles...", "clear add stack")
+
+#Save ROI's back to OMERO
```

Comment:
From where should the ROIs come ? Manual drawing in ImageJ ? Will the script suddenly become interactive ?

Reply:
Ah, sorry, of course, they come from the ``Analyze Particles...`` step.

Reply:
Actually , I tried now to simply draw some more ROIs manually on the image and save them to OMERO using the insight-ij plugin. But the plugin asks me about the images which were not imported to OMERO yet. Does it mean it does not recognize the image as coming from OMERO ?

Reply:
It would be great if the script could somehow pass on the insight-ij plugin the info about the fact that this image of ``user-x`` is already in OMERO and that what is needed is to add some ROIs to it from ImageJ.